doc/asciidoc/linkprediction-resource-allocation.adoc

```diff
@@ -0,0 +1,105 @@
+[[algorithms-linkprediction-resource-allocation]]
+= The Resource Allocation algorithm
+
+[abstract]
+--
+This section describes the Resource Allocation Similarity algorithm in the Neo4j Graph Algorithms library.
+--
+
+// tag::introduction[]
+link:https://arxiv.org/pdf/0901.0553.pdf[Resource Allocation similarity] is a measure used to compute similarity between nodes based on their shared neighbors.
+// end::introduction[]
+
+
+[[algorithms-linkprediction-resource-allocation-context]]
+== History and explanation
+
+// tag::explanation[]
+
+Resource Allocation similarity was introduced in 2009 by Tao Zhou, Linyuan Lü, and Yi-Cheng Zhang as part of a study to predict links in various networks.
+It is computed using the following formula:
+
+image::resource-allocation-similarity.svg[role="middle"]
+
+where `N(u)` is the set of nodes adjacent to `u`.
+
+A value of 0 indicates that two nodes are dissimilar, while higher values indicate nodes are more similar.
+
+The library contains a function to calculate similarity between two nodes.
+
+// end::explanation[]
+
+[[algorithms-linkprediction-resource-allocation-usecase]]
+== Use-cases - when to use the Resource Allocation Similarity algorithm
```

Comment:
This page has the same issue regarding consistency of name.

Here, we use **the Resource Allocation Similarity algorithm**, whereas the title is **the Resource Allocation algorithm**. I would suggest sticking to the titled version, unless there is a good reason not to, but consistency is best practice.

Reply:
I have updated this to remove 'Similarity' everywhere. I was originally going to include this algorithms in the 'similarity' category, but realised that the way they're used is completely different and it didn't make sense to have them there.